lib/vtls/schannel.c

```diff
@@ -1576,6 +1610,159 @@ CURLcode Curl_schannel_random(unsigned char *entropy, size_t length)
+
+  do {
+    status = s_pSecFn->QueryContextAttributes(&connssl->ctxt->ctxt_handle,
+                                              SECPKG_ATTR_REMOTE_CERT_CONTEXT,
+                                              &pCertContextServer);
+
+    if((status != SEC_E_OK) || (pCertContextServer == NULL)) {
+      failf(data, "schannel: Failed to read remote certificate context: %s",
+            Curl_sspi_strerror(conn, status));
+      break; /* failed */
+    }
+
+
+    if(!CryptAcquireContext(&hCryptProv, NULL, NULL,
+                            /*PROV_RSA_FULL,*/
+                            PROV_RSA_AES,
+                            CRYPT_VERIFYCONTEXT | CRYPT_SILENT)) {
+      failf(data, "schannel: Failed to acquire crypto context: %s",
+            Curl_sspi_strerror(conn, GetLastError()));
+      break; /* failed */
+    }
+
+    /* Get the public key information for the certificate.
+     This works for RSA keys, but secp256r1 and secp384r1 gives
+     this error:
+     Unknown error (0x8009310B) - ASN1 bad tag value met.
+     */
+    if(!CryptImportPublicKeyInfo(
+        hCryptProv,
+        X509_ASN_ENCODING,
+        &pCertContextServer->pCertInfo->SubjectPublicKeyInfo,
+        &hCertPubKey)) {
+      failf(data, "schannel: Failed to import public key info: %s",
+            Curl_sspi_strerror(conn, GetLastError()));
+      break; /* failed */
+    }
+
+    /* export to blob */
+    if(!CryptExportKey(hCertPubKey, 0, PUBLICKEYBLOB, 0, 0, &bloblen)) {
+      failf(data, "schannel: Failed to get public key blob length: %s",
+            Curl_sspi_strerror(conn, GetLastError()));
+      break; /* failed */
+    }
+
+    blob = malloc(bloblen);
+    if(!blob)
+      break; /* failed */
+
+    if(!CryptExportKey(hCertPubKey, 0, PUBLICKEYBLOB, 0, blob, &bloblen)) {
+      failf(data, "schannel: Failed to export public key to blob: %s",
+            Curl_sspi_strerror(conn, GetLastError()));
+      break; /* failed */
+    }
+
+    /* export to der 
+     not sure what second arg should be,
+     X509_PUBLIC_KEY_INFO -- fails
+     RSA_CSP_PUBLICKEYBLOB -- not quite der, missing spki header bytes
+     CNG_RSA_PUBLIC_KEY_BLOB -- fails
+     PKCS_CONTENT_INFO -- fails
+     X509_SEQUENCE_OF_ANY - fails
+     CRYPT_STRING_BASE64HEADER - fails
+     */
+    if(!CryptEncodeObjectEx(X509_ASN_ENCODING, RSA_CSP_PUBLICKEYBLOB, blob, 0,
+                            NULL, NULL, &pubkeylen)) {
+      failf(data, "schannel: Failed to get public key der length: %s",
+            Curl_sspi_strerror(conn, GetLastError()));
+      break;
+    }
+
+    pubkey = malloc(pubkeylen);
+    if(!pubkey)
+      break;
+
+    if(!CryptEncodeObjectEx(X509_ASN_ENCODING, RSA_CSP_PUBLICKEYBLOB, blob, 0,
+                            NULL, pubkey, &pubkeylen)) {
+      failf(data, "schannel: Failed to export public key to der: %s",
+            Curl_sspi_strerror(conn, GetLastError()));
+      break;
+    }
+
+    FILE *fp;
+    fp=fopen("windows.key", "wb");
```

Comment:
Yep again just debug code so it's easy to compare what we got out of windows with what we expect with tools like xxd